pisces/models.py

```diff
@@ -358,7 +401,17 @@ def run_splits(split_maker: SplitMaker,
     split_results = []
     splits = []
 
-    swc = swc_class(data_processor, epochs=epochs)
+    if swc_class == SGDLinearClassifier and not linear_model:
+        raise ValueError("Must provide a linear model for SGDLinearClassifier")
+    elif not swc_class == SGDLinearClassifier and linear_model:
+        raise ValueError("Linear model provided but not using SGDLinearClassifier")
+    elif swc_class == SGDLinearClassifier and linear_model:
```

Comment:
I think these 3 remaining cases that don't raise errors can be combined, since/once the subclass inits all have `**kwargs` at the end? so that `linear_model` will just get discarded for classes that don't care.